### PR TITLE
Update detekt, AGP, Gradle - and fix project formatting

### PR DESCRIPTION
**Pull request by elihart**
This updates our detekt implementation to use the official gradle plugin instead of a script. It also updates our detekt version to latest and fixes new rules as necessary.

Detekt is now run with `./gradlew detekt`. This automatically applies formatting autocorrection with ktlint. `./gradlew detekt -PCI=true` is used in our travis config to not use autocorrect on CI.

I have also updated gradle and AGP to latest (6.4.1 and 4.0).

Lastly, I have run intellij code formatting on the entire project.

I've split the commits up to make reviewing easier.

### CODE REVIEW

File: dogs/src/main/res/raw/dog.json [hunk 1]
```diff
@@ -1 +1,4212 @@
...
\ No newline at end of file
+{
```
Comment (gpeal) [inline dogs/src/main/res/raw/dog.json:1]: Can we have it ignore json files? This actually increases file size and time to parse.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.4.1-all.zip
```
Comment (gpeal) [inline gradle/wrapper/gradle-wrapper.properties:6]: 6.5 is out now :)

Comment (elihart) [reply gradle/wrapper/gradle-wrapper.properties:6]: yep! updated our airbnb repo today

File: dogs/src/main/res/raw/dog.json (hunk 1 above)
Comment (elihart) [reply dogs/src/main/res/raw/dog.json:1]: this wasn't detekt, this was intellij formatting applied to the whole project, so it won't normally run

Comment (gpeal) [reply dogs/src/main/res/raw/dog.json:1]: @elihart Can you revert this one?

**Comment by gpeal** [comment]
@elihart For some reason, one of the tests is failing now

**Pull request merged**